api/src/test/java/org/openmrs/api/db/hibernate/HibernateOrderDAOTest.java

```diff
@@ -63,6 +72,80 @@ public void saveOrderGroup_shouldSaveOrderGroup() {
 		for (Order savedOrder : savedOrderGroup.getOrders()) {
 			assertNull("Order is not saved as a part of Order Group", savedOrder.getOrderId());
 		}
-		
+	}
+	/**
+	 * @see {@link HibernateOrderDAO#getOrderGroupsByEncounter(Encounter)}
+	 * @throws Exception
+	 */
+	@Test(expected = APIException.class)
+	public void getOrderGroupsByEncounter_shouldFailGivenNullEncounter() {
+		dao.getOrderGroupsByEncounter(null);
+	}
+
+	/**
+	 * @see {@link HibernateOrderDAO#getOrderGroupsByPatient(Patient)}
+	 * @throws Exception
+	 */
+	@Test(expected = APIException.class)
+	public void getOrderGroupsByPatient_shouldFailGivenNullPatient() {
+		dao.getOrderGroupsByPatient(null);
+	}
+
+	/**
+	 * @see {@link HibernateOrderDAO#getOrderGroupsByEncounter(Encounter)}
+	 * @throws Exception
+	 */
+	@Test
+	public void getOrderGroupsByEncounter_shouldGetOrderGroupsFromAnEncounter() {
+		Encounter existingEncounter = Context.getEncounterService().getEncounter(4);
+		List<OrderGroup> ordergroups1 = Context.getOrderService().getOrderGroupsByEncounter(existingEncounter);
+
+		assertEquals(1, ordergroups1.size());
+	}
+
+	/**
+	 * @see {@link HibernateOrderDAO#getOrderGroupsByEncounter(Encounter)}
+	 * @throws Exception
+	 */
+	@Test
+	public void getOrderGroupsByEncounter_shouldGetOrderGroupsFrommMultipleEncounters() {
+		Encounter existingEncounter1 = Context.getEncounterService().getEncounter(3);
+		Encounter existingEncounter2 = Context.getEncounterService().getEncounter(6);
+		List<OrderGroup> ordergroups1 = Context.getOrderService().getOrderGroupsByEncounter(existingEncounter1);
+		List<OrderGroup> ordergroups2 = Context.getOrderService().getOrderGroupsByEncounter(existingEncounter2);
+
+		assertEquals(2, ordergroups1.size());
+		assertEquals(1, ordergroups2.size());
+	}
+
+	/**
+	 * @see {@link HibernateOrderDAO#getOrderGroupsByPatient(Patient)}
+	 * @throws Exception
+	 */
+	@Test
+	public void getOrderGroupsByPatient_shouldGetOrderGroupsFromMultiplePatients() {
+
+		Patient existingPatient1 = Context.getPatientService().getPatient(7);
+		Patient existingPatient2 = Context.getPatientService().getPatient(2);
```

Comment:
What is the use of this test?

Reply:
the purpose of this test is to create instances of multiple patients and they do exist in the existing code

Reply:
Can you point me to any tests that you found doing the same?

Reply:
your right @dkayiwa the tests are not there, am going to remove them and only maintain the tests relating to getting order groups from a patient